### PR TITLE
Success page after proposal creation

### PR DESCRIPTION
**Pull request by jparnaudo**
This PR addresses [this issue](https://www.notion.so/Success-Page-after-Proposal-Creation-fead327c947042909faa23e2a8bc5d25)

Follow Up modal with possible actions shows when the proposal url has the parameter `newProposal=true`.

Looks like this:
image

When the proposal link is copied, the button changes for 1.5s:
image

When visiting a proposal from the main section, Follow Up modal doesn't show

image

### CODE REVIEW

File: src/components/Modal/FollowUpModal.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  onDismiss: (e: React.MouseEvent<any>) => void,
+  forumUrl: string | null
+}
+
+export function FollowUpModal({ open, onDismiss, forumUrl, ...props }: FollowUpModalProps) {
+  const l = useFormatMessage()
+  const linkToProposal = window.location.href.split('&newProposal')[0]
+  const [isCopied, setIsCopied] = useState(false);
+
+  async function copyToClipboard(text: string) {
```
Comment (2fd) [inline src/components/Modal/FollowUpModal.tsx:25]: you have a `useClipboarCopy` in `decentraland-gatsby`

File: src/components/Section/SubscribeButton.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ export type SubscribeButtonProps = Omit<React.AnchorHTMLAttributes<HTMLAnchorEle
 
 export default React.memo(function SubscribeButton({ loading, disabled, subscribed, ...props }: SubscribeButtonProps) {
   const l = useFormatMessage()
-  return <a {...props} target="_black"  className={TokenList.join([
+  return <a {...props} target="_blank"  className={TokenList.join([
```
Comment (2fd) [inline src/components/Section/SubscribeButton.tsx:20]: Whenever you use `target="_blank"` you should use `rel="noopener noreferrer"`

File: src/modules/locations.ts [hunk 3]
```diff
@@ -76,9 +76,9 @@ export function url(path: string = '/', query: Record<string, string> | URLSearc
 
 export default {
   proposals: (options: Partial<ProposalListPage & ProposalListViewFilter & ProposalsStatusFilter & ProposalsTypeFilter & ProposalsModal> | URLSearchParams = {}) => url('/', options),
-  proposal: (proposal: string) => url(`/proposal/`, { id: proposal }),
+  proposal: (proposal: string, newProposal: boolean = false ) => url(`/proposal/`, { id: proposal, newProposal: newProposal.toString()}),
```
Comment (2fd) [inline src/modules/locations.ts:79]: extra params should be used as options and the shorter version is preferred

```suggestion
  proposal: (proposal: string, options: { new?: boolean } = {} ) => url(`/proposal/`, { id: proposal, ...options }),
```

File: src/pages/proposal.tsx [hunk 4]
```diff
@@ -42,20 +41,22 @@ import './index.css'
...
+  showVotesList: boolean,
+  showFollowUpModal: boolean
 }
 
 export default function ProposalPage() {
   const l = useFormatMessage()
   const location = useLocation()
   const params = useMemo(() => new URLSearchParams(location.search), [location.search])
-  const [options, patchOptions] = usePatchState<ProposalPageOptions>({ changing: false, confirmSubscription: false, confirmDeletion: false, confirmStatusUpdate: false, showVotesList: false })
+  const [options, patchOptions] = usePatchState<ProposalPageOptions>({ changing: false, confirmSubscription: false, confirmDeletion: false, confirmStatusUpdate: false, showVotesList: false, showFollowUpModal: params.get('newProposal') === "true" })
```
Comment (2fd) [inline src/pages/proposal.tsx:59]: for consistency all state derived from url should start as `false` and use a `useEffect` to detect any chance and update it

that way the update of that state came from the url change and not the `setState`

File: src/pages/proposal.tsx [hunk 5]
```diff
@@ -228,5 +229,11 @@ export default function ProposalPage() {
       onClickAccept={(_, status, description) => updateProposalStatus(status, description)}
       onClose={() => patchOptions({ confirmStatusUpdate: false })}
     />
+    <FollowUpModal
+      open={options.showFollowUpModal}
+      onDismiss={() => patchOptions({ showFollowUpModal: false })}
```
Comment (2fd) [inline src/pages/proposal.tsx:234]: for state derived from url its update should be also and url update

File: src/pages/submit/ban-name.tsx [hunk 6]
```diff
@@ -91,7 +91,7 @@ export default function SubmitBanName() {
         })
         .then((proposal) => {
             loader.proposals.set(proposal.id, proposal)
-            navigate(locations.proposal(proposal.id), { replace: true })
+            navigate(locations.proposal(proposal.id, true), { replace: true })
```
Comment (2fd) [inline src/pages/submit/ban-name.tsx:94]: use the options and shorter version

```suggestion
            navigate(locations.proposal(proposal.id,{ new: true }), { replace: true })
```

File: src/pages/submit/grant.tsx [hunk 7]
```diff
@@ -185,7 +185,7 @@ export default function SubmitBanName() {
         })
         .then((proposal) => {
             loader.proposals.set(proposal.id, proposal)
-            navigate(locations.proposal(proposal.id), { replace: true })
+            navigate(locations.proposal(proposal.id, true), { replace: true })
```
Comment (2fd) [inline src/pages/submit/grant.tsx:188]: use the options and shorter version

```suggestion
            navigate(locations.proposal(proposal.id,{ new: true }), { replace: true })
```

File: src/pages/submit/poi.tsx [hunk 8]
```diff
@@ -119,7 +119,7 @@ export default function SubmitPOI() {
         })
         .then((proposal) => {
           loader.proposals.set(proposal.id, proposal)
-          navigate(locations.proposal(proposal.id), { replace: true })
+          navigate(locations.proposal(proposal.id, true), { replace: true })
```
Comment (2fd) [inline src/pages/submit/poi.tsx:122]: use the options and shorter version

```suggestion
            navigate(locations.proposal(proposal.id,{ new: true }), { replace: true })
```

File: src/pages/submit/poll.tsx [hunk 9]
```diff
@@ -129,7 +129,7 @@ export default function SubmitPoll() {
         })
         .then((proposal) => {
           loader.proposals.set(proposal.id, proposal)
-          navigate(locations.proposal(proposal.id), { replace: true })
+          navigate(locations.proposal(proposal.id, true), { replace: true })
```
Comment (2fd) [inline src/pages/submit/poll.tsx:132]: use the options and shorter version

```suggestion
            navigate(locations.proposal(proposal.id,{ new: true }), { replace: true })
```

**Review by 2fd — CHANGES_REQUESTED** [review]

File: src/modules/locations.ts (hunk 3 above)
Comment (jparnaudo) [reply src/modules/locations.ts:79]: we are going with this definition `options: { new?: "true" } = {}` because the url function def expects strings

**Review by 2fd — APPROVED** [review]

**Pull request merged**